docs/TuxMutableClass.md

```diff
@@ -0,0 +1,201 @@
+
+Expected keys:
+
+###### <a id="mutableTraits-props"></a>Property - 'mutableTraits.props' - type: STRING, ARRAY of STRINGs, ARRAY of ARRAYs of STRINGs - optional [#](#mutableTraits-props)
+A trait, or array of traits, or an array of arrays of traits, within the components props object to create a path for. An example will follow the `mutableTraits.state` explanation.
+
+###### <a id="mutableTraits-state"></a>Property - 'mutableTraits.state' - type: STRING, ARRAY of STRINGs, ARRAY of ARRAYs of STRINGs - optional [#](#mutableTraits-state)
+A trait, or array of traits, or an array of arrays of traits, within the components state object to create a path for.
+
+```javascript
+    var mutableClassProps = {};
+    mutableClassProps.mutableTraits = {
+      props: 'text',
+      state: 'editing'
+    };
+
+    // OR
+    mutableClassProps.mutableTraits = {
+      props: ['text', 'timestamp'],
+      state: 'editing'
+    };
+
+    // OR it is possible to specify the hierarchy of keys to search through to improve performance of the one time deep search/provide greater specificity regarding the required property
+    mutableClassProps.mutableTraits = {
+      props: [
+        ['message', 'text'],
+        ['message', 'timestamp']
+      ],
+      state: 'editing'
+    };
+```
+
+#### <a id="mutableClassProps-render"></a>Parameter - `mutableClassProps.render` - type: FUNCTION - required [#](#createMutableClass-render)
+Since this is an implementation of React.createClass a render method is required.
+
+#### <a id="createMutableClass-mutableClass"></a>Return - 'mutableClass' - type: OBJECT [#](#createMutableClass-mutableClass)
+Returns a React class augmented with special Tux mixins.
+
+## <a id="TuxMutableClass-Guidelines"></a>TuxMutableClass Guidelines [#](#TuxMutableClass-Guidelines)
+We recommend using the mutable class in situations where you are rendering many components of the same class, and only some of the state and props in those components will ever update.
+
+A specific example is if you have tens or hundreds of message components rendering on the screen. The mutable class would be beneficial at that point because most of those message components can update, but will not necessarily update with great frequency. Therefore having components that will check to see if they should re-render in the virtual DOM can result in a performance boost with the more messages that are on the screen. If someone edits the text of a message, instead of re-rendering all of the other messages in the virtual DOM with the mutable class only the one message that has updated will do so.
+
+```javascript
+    // GOOD
+
+
+
+    // BAD
+```
+
+Taking all that we have learned so far let's take a look at a more realistic example.
+
+## <a id="TuxMutableClass-Complete-Example"></a>TuxMutableClass Complete Example [#](#TuxMutableClass-Complete-Example)
```

Comment:
This example is a little complicated right now. Consider trimming down the complexity of the component and maybe remove the messageForm and just change it to an input field so there is less going on.